crates/holochain/src/conductor/conductor.rs

```diff
@@ -838,12 +841,120 @@ mod network_impls {
 
         pub(crate) async fn network_info(
             &self,
-            dnas: &[DnaHash],
+            payload: &NetworkInfoRequestPayload,
         ) -> ConductorResult<Vec<NetworkInfo>> {
+            use holochain_sqlite::sql::sql_cell::SUM_OF_RECEIVED_BYTES_SINCE_TIMESTAMP;
+
+            let NetworkInfoRequestPayload {
+                agent_pub_key,
+                dnas,
+                last_time_queried,
+            } = payload;
+
             futures::future::join_all(dnas.iter().map(|dna| async move {
-                let d = self.holochain_p2p.get_diagnostics(dna.clone()).await?;
-                let fetch_pool_info = d.fetch_pool.info([dna.to_kitsune()].into_iter().collect());
-                ConductorResult::Ok(NetworkInfo { fetch_pool_info })
+                let diagnostics = self.holochain_p2p.get_diagnostics(dna.clone()).await?;
+                let fetch_pool_info = diagnostics
+                    .fetch_pool
+                    .info([dna.to_kitsune()].into_iter().collect());
+
+                // query number of agents from peer db
+                let db = { self.p2p_agents_db(dna) };
+                let permit = db.conn_permit().await;
+                let mut conn = db.with_permit(permit)?;
+                let agent_infos = conn.p2p_list_agents()?;
+                let current_number_of_peers = agent_infos.len() as u32;
```

Comment:
This is pretty expensive, it decodes all the agent infos from the whole database to just get a count of them.

Reply:
Ah, right. Initially I thought it was better to reuse one of the existing calls, but didn't think of the consequences. I'd rather add another query then that returns a count. Should I make it one of the `p2p_` calls?

Reply:
Probably, but I'm not attached if it seems to fit better in some other way.

Reply:
How about this?

Reply:
lgtm